src/arrayMethodSort.js

```diff
@@ -5,8 +5,46 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let compFunc;
+    let hadChanges;
+
+    if (arguments.length === 0) {
+      compFunc = defaultCompareFunction;
+    } else {
+      compFunc = compareFunction;
```

Comment:
You can make it with ternary operator

`const compFunc = (arguments.length === 0) ? defaultCompareFunction : ........`

Reply:
actually I've learned another idea how to make this working, adding default function as a default value to function parameter